### ISSUE TITLE
Promt showing on wrong monitor

### ISSUE THREAD

**Issue by nhagen** [prikhi/lightdm-mini-greeter#10]
I'm not able to get the prompt to show up on my primary monitor, even after following the
recommendation [here](https://wiki.archlinux.org/index.php/LightDM#LightDM_displaying_in_wrong_monitor)

Is anyone else able to reproduce this?

**Comment by prikhi** [comment]
What's your monitor layout like? And which of those monitor do you want to the prompt to show up on?

I've got 3 monitors w/ the prompt on the center one, and something like this in `10-monitor.conf` in my `/etc/X11/xorg.conf.d` directory works for me:

```
Section "Monitor"
    Identifier "DP-2"
    Option "PreferredMode" "1050x1680"
    Option "DPMS"
    Option "Rotate" "left"
EndSection
Section "Monitor"
    Identifier "DVI-I-1"
    Option "PreferredMode" "2560x1440"
    Option "DPMS"
EndSection
Section "Monitor"
    Identifier "DP-0"
    Option "PreferredMode" "2560x1440"
    Option "Primary" "true"
    Option "DPMS"
EndSection
Section "Screen"
    Identifier "Screen0"
    DefaultDepth 24
    Option "metamodes" "DP-0: nvidia-auto-select +2560+0, DVI-I-1: nvidia-auto-select +0+0, DP-2: nvidia-auto-select +5120+0 {rotation=left}"
    SubSection "Display"
        Depth 24
    EndSubSection
EndSection
```

I haven't tried the xrandr setup script way though.

**Comment by prikhi** [comment]
also what's the xrandr command you're using in the config, and what's the output of `xrandr -q` when you're logged in?

**Comment by x0rzist** [comment]
I have the same problem. The 10-monitor.conf includes:
```
Section "Monitor"
    Identifier  "HDMI-A-0"
    Option      "Primary" "true"
EndSection

Section "Monitor"
    Identifier  "HDMI-A-1"
    Option      "Above" "HDMI-A-0"
EndSection

Section "Monitor"
    Identifier  "DVI-D-0"
    Option      "RightOf" "HDMI-A-0"
EndSection
```
So the prompt should show up on `HDMI-A-0`, but instead it's displayed on `DVI-D-0`. The output of `xrandr -q` is:

```
Screen 0: minimum 320 x 200, current 3840 x 2160, maximum 16384 x 16384
HDMI-A-0 connected primary 1920x1080+0+1080 (normal left inverted right x axis y axis) 598mm x 336mm
   1920x1080     60.00*+  71.91    50.00    59.94  
   1680x1050     59.88  
   1600x900      60.00  
   1280x1024     60.02  
   1440x900      59.90  
   1280x800      59.91  
   1280x720      60.00    50.00    59.94  
   1024x768      70.07    60.00  
   800x600       72.19    60.32    56.25  
   720x576       50.00  
   720x480       60.00    59.94  
   640x480       72.81    66.67    60.00    59.94  
   720x400       70.08  
DisplayPort-0 disconnected (normal left inverted right x axis y axis)
DisplayPort-1 disconnected (normal left inverted right x axis y axis)
HDMI-A-1 connected 1920x1080+0+0 (normal left inverted right x axis y axis) 1280mm x 720mm
   1920x1080     60.00*+  50.00    59.94    30.00    25.00    24.00    29.97    23.98  
   1920x1080i    60.00    50.00    59.94  
   1600x1200     60.00  
   1680x1050     59.88  
   1400x1050     59.95  
   1280x1024     60.02  
   1440x900      59.90  
   1280x960      60.00  
   1280x800      59.91  
   1280x720      60.00    50.00    59.94  
   1024x768      60.00  
   800x600       60.32  
   720x576       50.00  
   720x576i      50.00  
   720x480       60.00    59.94  
   720x480i      60.00    59.94  
   640x480       60.00    59.94  
DVI-D-0 connected 1920x1080+1920+1080 (normal left inverted right x axis y axis) 477mm x 268mm
   1920x1080     60.00*+
   1680x1050     59.88  
   1600x900      60.00  
   1280x1024     75.02    60.02  
   1280x960      60.00  
   1152x864      75.00  
   1280x720      60.00  
   1152x720      59.97  
   1024x768      75.03    60.00  
   832x624       74.55  
   800x600       75.00    60.32  
   640x480       75.00    59.94  
   720x400       70.08  
```

**Comment by nhagen** [comment]
Sorry, the machine that I was having this issue is out of commission for unrelated reasons so I'm not able to provide more debugging right now. Will keep following this thread in case I am able to provide any more information.

**Comment by prikhi** [comment]
@x0rzist Thanks, that helps. I've replicated your setup in virtualbox, gotten the main window to show up on the primary monitor, but still working on getting it centered correctly.

**Comment by prikhi** [comment]
Should be fixed by 53b21abfd429b8b758b89f89b2ca18407eb26383, releasing v0.3.2 now, please re-open if it doesn't solve the issue for you.
